lib/widget/staggered_grid.dart

```diff
@@ -82,29 +86,26 @@ class EnsembleStaggeredGridState extends WidgetState<EnsembleStaggeredGrid>
   Widget buildWidget(BuildContext context) {
     final items = buildItems();
 
-    return SingleChildScrollView(
-      child: StaggeredGrid.count(
-        crossAxisCount: 4,
-        mainAxisSpacing: widget._controller.verticalGap ?? 0,
-        crossAxisSpacing: widget._controller.horizontalGap ?? 0,
-        children: List.generate(items.length, (index) {
-          final item = items[index];
-
-          return StaggeredGridTile.count(
-            crossAxisCellCount: 2,
-            mainAxisCellCount: getRandomMainAxisCount,
-            child: GestureDetector(
-              onTap: () => _onItemTapped(index),
-              child: item,
-            ),
-          );
-        }),
-      ),
+    return StaggeredGridView.countBuilder(
+      crossAxisCount: widget._controller.crossAxisCount ?? 4,
+      mainAxisSpacing: widget._controller.verticalGap ?? 0,
+      crossAxisSpacing: widget._controller.horizontalGap ?? 0,
+      padding: widget._controller.padding,
+      itemCount: items.length,
+      staggeredTileBuilder: (index) => _staggeredTiles[index],
+      itemBuilder: (context, index) {
+        final item = items[index];
+
+        return GestureDetector(
+          onTap: () => _onItemTapped(index),
```

Comment:
I know we do this elsewhere too, but the user won't know whether the index comes from children or item-template and have to do manual calculations. Can we enforce that either 'children' or 'item-template' can be used at a time. Using both will throw a LanguageError.

Reply:
I meant you can use either children or item-template, but not both. Error will be throw if both are specified.

Reply:
Okay, I updated it now